File: actor-simulator/src/main/java/at/tuwien/dse/actorsimulator/dto/Movement.java
package at.tuwien.dse.actorsimulator.dto;

import java.time.LocalDateTime;

public class Movement {

    private String vin;
    private double speed;
    private LocalDateTime dateTime;
    private double longitude;
    private double latitude;
    private boolean crash;

    public String getVin() {
        return vin;
    }

    public void setVin(String vin) {
        this.vin = vin;
    }

    public double getSpeed() {
        return speed;
    }

    public void setSpeed(double speed) {
        this.speed = speed;
    }

    public LocalDateTime getDateTime() {
        return dateTime;
    }

    public void setDateTime(LocalDateTime dateTime) {
        this.dateTime = dateTime;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public boolean isCrash() {
        return crash;
    }

    public void setCrash(boolean crash) {
        this.crash = crash;
    }
}

File: client/src/app/models/movement.ts
import { Vehicle } from './vehicle';

export class Movement {
    vin: string;
    speed: number;
    longitude: number;
    latitude: number;
    dateTime: Date;
    crash: boolean;
    vehicle: Vehicle;

    constructor(vin: string, speed: number, longitude: number, latitude: number, crash: boolean){
        this.vin = vin;
        this.speed = speed;
        this.longitude = longitude;
        this.latitude = latitude;
        this.crash = crash;
    }

    fullInfo(){
        return 'VIN: ' + this.vin + '\n'
                + 'Model: ' + this.vehicle.model + '\n'
                + 'Producer: ' + this.vehicle.producer + '\n'
                + 'Speed: ' + this.speed + 'km/h \n'
                + 'Longitude: ' + this.longitude + '\n'
                + 'Latitude: ' + this.latitude + '\n'
                + 'Crash happened: ' + this.crash + '\n'
                
    }
}
